lib/rules/arrow-parens.js

```diff
@@ -74,35 +74,65 @@ module.exports = {
 
         const sourceCode = context.getSourceCode();
 
+        /**
+         * Checks if token is part of the given node.
+         * @param {ASTNode} token A node to check.
+         * @param {ASTNode} node The arrow function node.
+         * @returns {boolean} Whether or not the token is part of node.
+         */
+        function isTokenPartOfNode(token, node) {
+            return token.range[0] >= node.range[0] && token.range[1] <= node.range[1];
+        }
+
         /**
          * Determines whether a arrow function argument end with `)`
          * @param {ASTNode} node The arrow function node.
          * @returns {void}
          */
         function parens(node) {
-            const isAsync = node.async;
-            const firstTokenOfParam = sourceCode.getFirstToken(node, isAsync ? 1 : 0);
+            const arrowToken = sourceCode.getTokenBefore(node.body, astUtils.isArrowToken);
+
+            const openingParenToken = node.params.length >= 1 && node.params[0].type === "Identifier"
+                ? sourceCode.getTokenBefore(node.params[0], t => isTokenPartOfNode(t, node) && astUtils.isOpeningParenToken(t))
+                : sourceCode.getTokenBefore(arrowToken, t => isTokenPartOfNode(t, node) && astUtils.isOpeningParenToken(t));
```

Comment:
This could find an opening paren in default values for params.

Maybe we should just always get the first token before `params[0]`, and make sure that the rule works only on single-param arrow function by changing the `ArrowFunctionExpression` selector to: 

`"ArrowFunctionExpression[params.length=1]"`